diff -up openwsman-2.2.3/etc/pam/openwsman.orig openwsman-2.2.3/etc/pam/openwsman
--- openwsman-2.2.3/etc/pam/openwsman.orig	2009-08-31 07:33:35.000000000 +0200
+++ openwsman-2.2.3/etc/pam/openwsman	2017-01-18 13:15:53.770889562 +0100
@@ -1,7 +1,7 @@
 #%PAM-1.0
-auth       required	pam_unix2.so	nullok
+auth       required	pam_unix.so	nullok
 auth       required	pam_nologin.so
-account    required	pam_unix2.so
-password   required	pam_pwcheck.so	nullok
-password   required	pam_unix2.so	nullok use_first_pass use_authtok
-session    required	pam_unix2.so	none
+account    required	pam_unix.so
+password   required	pam_cracklib.so
+password   required	pam_unix.so	nullok use_first_pass use_authtok
+session    required	pam_unix.so
